src/power_grid_model/validation/rules.py

```diff
@@ -697,6 +698,31 @@ def none_missing(data: SingleDataset, component: str, fields: Union[str, List[st
     return errors
 
 
+def valid_p_q_sigma(data: SingleDataset, component: str) -> List[MultiFieldValidationError]:
+    """
+    Check validity of the pair `(p_sigma, q_sigma)` for 'sym_power_sensor' and 'asym_power_sensor'.
+
+    Args:
+        data: The input/update data set for all components
+        component: The component of interest, in this case only 'sym_power_sensor' or 'asym_power_sensor'
+
+    Returns:
+        A list containing zero or one MultiFieldValidationError, listing the p_sigma and q_sigma mismatch
+    """
+    errors = []
+    p_sigma = data[component]["p_sigma"]
+    q_sigma = data[component]["q_sigma"]
+    ids = [data[component]["id"].flatten().tolist()]
+    p_nan = np.isnan(p_sigma)
+    q_nan = np.isnan(q_sigma)
+    p_inf = np.isinf(p_sigma)
+    q_inf = np.isinf(q_sigma)
+    mis_match = p_nan != q_nan  # same logic as np.logical_xor(p_nan.any(axis=-1), q_nan.any(axis=-1)).any()
```

Comment:
this is not sufficient for asymmetric components, because in this case, there might be some component that has e.g. `p_sigma = [nan, 1.0, nan], q_sigma = [nan, 1.0, nan]`. Your logic will yield `any([False, False, False]) = False` but the result should be `True` (mismatch).

That is due to the fact that `power_sigma = sqrt(((p_sigma[0] + p_sigma[1] + p_sigma[2])/3)^2 + ((q_sigma[0] + q_sigma[1] + q_sigma[2])/3)^2)` requires all elements to be not nan

In addition, please also verify that it works as intended when there are multiple components (only invalid components are flagged)